src/client/dfs/duns.c

```diff
@@ -246,9 +246,10 @@ duns_resolve_lustre_path(const char *path, struct duns_attr_t *attr)
 #endif
 
 #define UUID_REGEX "([a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}){1}"
-#define DAOS_FORMAT "^daos://"UUID_REGEX"/"UUID_REGEX"[/]?"
-#define DAOS_FORMAT_NO_PREFIX "^[/]+"UUID_REGEX"/"UUID_REGEX"[/]?"
-#define DAOS_FORMAT_NO_CONT "^daos://"UUID_REGEX"[/]?$"
+#define LABEL_REGEX "([a-zA-Z0-9._:]{1,256})"
```

Comment:
If this is the format definition of what we can accept as labels then it seems it should be in a daos header and widely accessible rather than in duns.

A daos_is_label_valid() function would be a good thing to have in many places.

Reply:
agreed. right now there isn't. i think many PRs are in the works. we can merge into a single usage once all land in a separate PR.